mutation-testing/output-pr-mutants/action.yml

```diff
@@ -0,0 +1,166 @@
+name: Output Mutants
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Download artifacts
+      uses: actions/download-artifact@v3
+
+    - name: Append output from shards
+      shell: bash
+      run: |
+        # List of all possible mutant outcomes file names and folder names collected from jobs.
+        folders=("mutants-shard-big--1" "mutants-shard-big-0" "mutants-shard-big-1" "mutants-shard-big-2" "mutants-shard-big-3" "mutants-shard-big-4" "mutants-shard-big-5" "mutants-shard-big-6" "mutants-shard-big-7" "mutants-shard-small--1" "mutants-shard-small-0" "mutants-shard-small-1" "mutants-shard-small-2" "mutants-shard-small-3")
+        files=("missed.txt" "caught.txt" "timeout.txt" "unviable.txt")
+        mkdir -p mutants-shards
+
+        # If the folder/file path exists, append the output to it's corresponding file name in a newly created folder that will contain all outputs combined
+        for file in "${files[@]}"; do
+          for folder in "${folders[@]}"; do
+            if [[ -s "$folder/$file" ]]; then
+              cat "$folder/$file" >> "mutants-shards/$file"
+            fi
+          done
+        done
+
+        # If the folder/exit_code.txt path exists, check for the exit code and retain the most relevant one to a file
+        # - 4: unmutated build failed
+        # - 1: incorrect command line arguments
+        # - 2, 3: found timeout/missed/unviable mutants
+        # - 0: everything worked fine
+        # - *: unknown exit code
+        for folder in "${folders[@]}"; do
+          if [[ -s "$folder/exit_code.txt" ]]; then
+            exit_code=$(<"${folder}/exit_code.txt")
+            most_relevant_exit_code=0
+
+            case $exit_code in
+            4)
+              most_relevant_exit_code=4
+              ;;
+            1)
+              [ "$most_relevant_exit_code" -eq 0 ] && most_relevant_exit_code=1
+              ;;
+            2)
+              [ "$most_relevant_exit_code" -eq 0 ] && most_relevant_exit_code=2
+              ;;
+            3)
+              [ "$most_relevant_exit_code" -eq 0 ] && most_relevant_exit_code=3
+              ;;
+            0)
+              ;;
+            *)
+              echo "Unknown exit code $exit_code"
+              most_relevant_exit_code=$exit_code
+              ;;
+            esac
+          fi
```

Comment:
what should happen if the file `$folder/exit_code.txt` does not exist?
should there be a default exit code maybe?